pkgs/applications/editors/emacs/elisp-packages/manual-packages/codeium/default.nix

```diff
@@ -0,0 +1,35 @@
+{
+  trivialBuild,
+  fetchFromGitHub,
+  pkgs,
+  lib,
+}:
+trivialBuild {
+  pname = "codeium";
+  version = "1.6.13";
+  src = fetchFromGitHub {
+    owner = "Exafunction";
+    repo = "codeium.el";
+    rev = "1.6.13";
+    hash = "sha256-CjT21GhryO8/iM0Uzm/s/I32WqVo4M3tSlHC06iEDXA=";
+  };
+
+  buildInputs = [ pkgs.codeium ];
```

Comment:
It's better to get this from the function argument above.

Reply:
Yes, I have tried it. but I got a error.

```
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'emacs-codeium-1.6.13'
         whose name attribute is located at /home/grass/workspace/forks/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'buildInputs' of derivation 'emacs-codeium-1.6.13'

         at /home/grass/workspace/forks/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:395:7:

          394|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          395|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          396|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       error: infinite recursion encountered

       at «none»:0: (source not available)
```


I'm not sure why, so I didn't do it.


I guess it's because it overlaps with the top-level package name, If you could help me, that would be even better

Reply:
It didn't cross my mind that they would collide.
You could change what `callPackage` injects as `codeium`, but given the naming collision I actually prefer it the way you've done it already because it's more clear.